Hestia/app/src/main/java/hestia/UI/activities/home/HomeActivity.java

```diff
@@ -79,14 +80,15 @@ protected void onDestroy() {
     }
 
     private void storeIP() {
-        SharedPreferences prefs = getSharedPreferences(HESTIA_IP, 0);
-        serverCollectionsInteractor.getHandler().setIp(prefs.getString(SERVER_IP, serverCollectionsInteractor.getHandler().getIp()));
+        SharedPreferences prefs = getSharedPreferences(HESTIA_IP, Context.MODE_PRIVATE);
+        prefs.edit().putString(SERVER_IP,serverCollectionsInteractor.getHandler().getIp()).apply();
     }
 
     private void setupCache() {
-        SharedPreferences prefs = getSharedPreferences(HESTIA_IP, 0);
-        String ip = prefs.getString(SERVER_IP, "192.168.178.31");
-        NetworkHandler handler = new NetworkHandler(ip, 8000);
+        SharedPreferences prefs = getSharedPreferences(HESTIA_IP, Context.MODE_PRIVATE);
+        String ip = prefs.getString(SERVER_IP, getApplicationContext().getString(R.string.default_ip));
+        NetworkHandler handler = new NetworkHandler(ip, Integer.valueOf(
```

Comment:
it would look nicer if you would store the port number in a separate variable and then pass it in the NH constructor.